### ISSUE TITLE
Train depots

### ISSUE THREAD

**Issue by gpcf** [orwell96/advtrains#16]
Here are a couple of suggestions, which I may implement in the next days (if I have time). These come from my experiences with running this mod on my server (I'm still running an old version, as the new ones need Minetest 0.4.15). 

Add an owner to the rails, so they are protected automatically, since it is a pain to protect all tracks. It should also protect the surroundings automatically, to prevent blocking. Also, some portions of track should be made public for other players to connect.

Also, I would add simutrans-style depots to minetest, because people tend to place trains on the tracks and they can't be removed.

**Comment by Desour** [comment]
maybe a tool for rail protecting, switching between protected and not
or make the locked rails protect a line or an amount of rails that are connected to the protected one

i like the idea of simutrans style depots but orwell said [here](https://forum.minetest.net/viewtopic.php?f=11&t=14726&start=125#p236731)
> Do not make trains craftable. Having trains in your inventory is strange. Rather add a depot to build the trains and they are placed on the tracks automatically:
not this way, but as soon as you place a wagon, you can't get its contents back, this should be realistic enough for the moment.

**Comment by gpcf** [comment]
I suggested the depot Idea in the forum a while ago, and I'm sure that it's a good idea now, not because of the realism, but because you don't want to have stray trains on the tracks. I'll see if I can program it in the next few days.

**Comment by orwell96** [comment]
Owner for every rail node? On most servers (e.g. cash's world) they manage to protect the roads without this. the protection logos are not walkable, you can place them right on rails.
Depots?! Please explain to me how they work in detail.
limiting access to seats: soon

**Comment by Desour** [comment]
depots are buildings which place a train onto a track

**Comment by gpcf** [comment]
I use areas on my server, the problem is that people build rails a lot faster than they manage to protect. I'll think of ideas on how to solve this. I would add a depot-building block, that builds an entire depot when placed, complete with tracks and all. Trains would be built there and roll on from there onto the tracks.

**Comment by orwell96** [comment]
yeah, areas (or anything that works well with big cuboids linke worldedit) is pretty useless on my rails
maybe use these node based protectors
